src/core/src/core_logic/PatchInstaller.py

```diff
@@ -468,7 +529,11 @@ def install_packages_in_batches(self, all_packages, packages, package_versions,
 
             per_batch_installation_stopwatch.stop_and_write_telemetry(str(per_batch_install_perf_log))
 
-        return installed_update_count, patch_installation_successful, maintenance_window_batch_cutoff_reached
+        # Performing reconciliation at the end to get accurate number of installed packages through this function.
+        installed_update_count += self.perform_status_reconciliation_conditionally(package_manager, True)
+        not_attempted_and_failed_packages = remaining_packages + failed_packages
```

Comment:
Making a list containing not attempted packages and failed packages in batch patching process. In fallback to sequential patching, this list is used as the input packages to install. The method get_remaining_packages_to_install is no longer required. But due to this change, the newly available packages in brief time while batch patching was done are not handled. This was the case before this PR also, the newly added packages while installing the packages were not handled. To make it simple, it is better to not handle that case in this big PR. It is difficult to test this scenario.

Reply:
Preparing single list not_attempted_and_failed_packages containing of two kind of packages:
(a) Not attempted due to not enough time in maintenance window to install packages in batches.
(b) Failed to install in batch patching.

These packages are attempted in the sequential patching if there is enough time remaining in maintenance window. The non attempted packages are in the front of the list and hence they are attempted first in sequential patching than the failed packages.

Will add comment for this.